spec: propose an evolution of closeOnExit (#2039)
microsoft/terminal: doc/specs/#xxxx - closeOnExit evolution.md

---
author: Dustin Howett @DHowett-MSFT
created on: 2019-07-19
last updated: 2019-07-19
issue id: "#xxxx"
---

# Improvements to CloseOnExit

## Abstract

This spec describes an improvement to the `closeOnExit` profile feature that will offer greater flexibility and allow us to provide saner defaults.

## Inspiration

Other terminal emulators like ConEmu have a similar feature.

## Solution Design

1. The existing `closeOnExit` profile key will become an enumerated string key with the following values/behaviours:
   * `always` - a tab or pane hosting this profile will always be closed when the launched connection terminates.

>    * `graceful` - a tab or pane hosting this profile will be closed **iff** the launched connection closes with an exit code == 0.

Reviewer: `iff`?

>    * `graceful` - a tab or pane hosting this profile will only be closed **if** the launched connection closes with an exit code != 0.

Author: ```suggestion
   * `graceful` - a tab or pane hosting this profile will be closed **iff** the launched connection closes with an exit code == 0.
```
